service/ec2/customizations.go

```diff
@@ -8,7 +8,43 @@ import (
 	"github.com/aws/aws-sdk-go-v2/internal/awsutil"
 )
 
+type retryer struct {
+	aws.DefaultRetryer
+}
+
+func (d retryer) RetryRules(r *request.Request) time.Duration {
+	switch r.Operation.Name {
+	case opModifyNetworkInterfaceAttribute:
+		fallthrough
+	case opAssignPrivateIpAddresses:
+		return d.customRetryRule(r)
+	default:
+		return d.DefaultRetryer.RetryRules(r)
+	}
+}
+
+func (d retryer) customRetryRule(r *request.Request) time.Duration {
+	d.MinTime = 1000
+	return d.DefaultRetryer.RetryRules(r)
+}
+
+func setCustomRetryer(c *Client) {
+	maxRetries := 3
+	c.Retryer = retryer{
+		DefaultRetryer: request.DefaultRetryer{
+			NumMaxRetries: maxRetries,
+		},
+	}
+}
+
 func init() {
+	initClient = func(c *Client) {
+		if c.Config.Retryer == nil {
+			// Only override the retryer with a custom one if the config
+			// does not already contain a retryer
+			setCustomRetryer(c)
```

Comment:
I think it would be better for `setCustomRetryer` return a `retryer` instead of setting the `Retryer` field within its own function.  (Also would be good example of using the constant suggested above for `DefaultRetryerMaxNumRetries`)

```go
if c.Config.Retryer == nil {
    c.Retryer = retryer{
         DefaultRetryer: request.DefaultRetryer{
              NumMaxRetries: 3,
              MinTime: 30,
         },
    }
}
```